### PR TITLE
Refactor GitHub Action Workflows

### PR DESCRIPTION
**Pull request by Supereg**
## :recycle: Current situation

Currently, the project contains workflow files that don't reuse the organization wide workflow files (see https://github.com/homebridge/.github).
The beta branch currently contains both, the standard workflow templates and the legacy workflow files of this project.

## :bulb: Proposed solution

This PR refactors the workflow files to:
* Build test and lint using the reusable workflow file (e.g., centrally defining supported node versions)
* Create an automated release and beta-release workflow
* Add a PR auto labeler as in all the other homebridge based repositories
* Add wiki change notifications

## :gear: Release Notes

* Improved the automated CI/CD pipeline of the project.

## :heavy_plus_sign: Additional Information

This PR creates a new workflow for managing releases, creating a similar workflow as for any other repository.
* Use `npm version <increment>` command to bump the version, create a version commit and create a version tag
* Push the changes and the git tag. This will trigger the `release.yml` workflow and publishes a new release to npm
* Now create a new release on GitHub. Ideally use the "autogenerate" option provided by the GitHub UI
* The `attach-artifact.yml` workflow is triggered and attaches the artifacts to the release

Similar beta release are automatically published for every push to a `beta-*.*.*` branch.

### Testing

--

### Reviewer Nudging

--

### CODE REVIEW

**Comment by Supereg** [comment]
Seems like the hb_service tests are failing for ubuntu-latest. Looking at the other PRs, this seems to be also an issue on the main branch.
@donavanbecker any ideas what the issue might be?

**Comment by donavanbecker** [comment]
I don't, I haven't been able to look into it.

**Review by donavanbecker — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -11,13 +11,11 @@ jobs:
...
       fail-fast: false
       matrix:
-        os: [ ubuntu-latest, macos-latest, windows-2019 ]
+        os: [ubuntu-latest, macos-latest, windows-2019]
 
     uses: homebridge/.github/.github/workflows/nodejs-build-and-test.yml@main
     with:
       enable_coverage: true
-      runs_on: ${{ matrix.os }}
-      install_cmd: npm ci && cd ui && npm ci
```
Comment (Supereg) [inline .github/workflows/build.yml]: What are these changes, they break the workflow run

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -68,11 +78,14 @@ jobs:
...
+        if: ${{ failure() }}
+        run: |
+          node dist/bin/hb-service.js view
+          exit 1
 
   lint:
+    needs: build_and_test
     uses: homebridge/.github/.github/workflows/eslint.yml@main
-    with:
-      install_cmd: npm ci && cd ui && npm ci
```
Comment (Supereg) [inline .github/workflows/build.yml]: Same here.

File: .github/workflows/attach-artifact.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    steps:
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: 16.x
+
+      # Sanity check to ensure that release tags don't start with a 'v' version prefix but adhere to the X.Y.Z format
+      - name: Check for Tag name Format
+        if: startWith(${{github.event.release.tag_name}}, 'v')
```
Comment (NorthernMan54) [inline .github/workflows/attach-artifact.yml:20]: Something is broke here

```
##[debug]Evaluating condition for step: 'Check for Tag name Format'
[2](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:2)
##[debug]Evaluating: (success() && format('startWith({0}, ''v'')', github.event.release.tag_name))
[3](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:3)
##[debug]Evaluating And:
[4](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:4)
##[debug]..Evaluating success:
[5](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:5)
##[debug]..=> true
[6](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:6)
##[debug]..Evaluating format:
[7](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:7)
##[debug]....Evaluating String:
[8](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:8)
##[debug]....=> 'startWith({0}, ''v'')'
[9](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:9)
##[debug]....Evaluating Index:
[10](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:10)
##[debug]......Evaluating Index:
[11](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:11)
##[debug]........Evaluating Index:
[12](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:12)
##[debug]..........Evaluating github:
[13](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:13)
##[debug]..........=> Object
[14](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:14)
##[debug]..........Evaluating String:
[15](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:15)
##[debug]..........=> 'event'
[16](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:16)
##[debug]........=> Object
[17](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:17)
##[debug]........Evaluating String:
[18](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:18)
##[debug]........=> 'release'
[19](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:19)
##[debug]......=> Object
[20](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:20)
##[debug]......Evaluating String:
[21](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:21)
##[debug]......=> 'tag_name'
[22](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:22)
##[debug]....=> '4.50.3'
[23](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:23)
##[debug]..=> 'startWith(4.50.3, ''v'')'
[24](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:24)
##[debug]=> 'startWith(4.50.3, ''v'')'
[25](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:25)
##[debug]Expanded: (true && 'startWith(4.50.3, ''v'')')
[26](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:26)
##[debug]Result: 'startWith(4.50.3, ''v'')'
[27](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:27)
##[debug]Starting: Check for Tag name Format
[28](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:28)
##[debug]Loading inputs
[29](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:29)
##[debug]Loading env
[30](https://github.com/homebridge/homebridge-config-ui-x/actions/runs/4947745386/jobs/8857592466#step:3:30)

File: .github/workflows/build.yml (hunk 1 above)
Comment (NorthernMan54) [reply .github/workflows/build.yml]: Adding back with 4.50.4

File: .github/workflows/build.yml (hunk 2 above)
Comment (NorthernMan54) [reply .github/workflows/build.yml]: Adding back with 4.50.4

**Pull request merged**